packages/gatsby/src/bootstrap/load-plugins/index.ts

```diff
@@ -45,10 +51,35 @@ const flattenPlugins = (plugins: Array<IPluginInfo>): Array<IPluginInfo> => {
   return flattened
 }
 
+const normalizePlugins = (config: IRawSiteConfig = {}): ISiteConfig => {
+  return {
+    ...config,
+    plugins: (config.plugins || []).map(
+      (plugin): IPluginRefObject => {
+        if (typeof plugin === `string`)
+          return {
+            resolve: plugin,
+            options: {},
+          }
+
+        return plugin
+      }
+    ),
+  }
```

Comment:
Also normalize subplugins

```suggestion
  return {
    ...config,
    plugins: (config.plugins || []).map(
      (plugin): IPluginRefObject => {
        if (typeof plugin === `string`) {
          return {
            resolve: plugin,
            options: {},
          }
        }

        if (plugin.options.plugins) {
          plugin.options = normalizePlugins(plugin.options)
        }

        return plugin
      }
    ),
  }
```